inference-engine/src/low_precision_transformations/src/layer_transformation.cpp

```diff
@@ -74,17 +74,35 @@ bool LayerTransformation::canBeTransformed(const TransformationContext& context,
         return false;
     }
 
+    if (NetworkHelper::isDQByDynamicDimension(layer)) {
+        return false;
+    }
+
     for (const auto& output : layer->outputs()) {
-        const size_t size = output.get_shape().size();
-        if ((size < 2ul) || (size > 5ul)) {
+        const auto rank = output.get_partial_shape().rank();
+        if (rank.is_dynamic()) {
+            return false;
+        }
+        auto size = rank.get_length();
+        if ((size < 2) || (size > 5)) {
             return false;
         }
     }
 
     const auto dequantization = NetworkHelper::getDequantization(layer);
     if (!dequantization.empty()) {
-        auto perChannelQuantization = [](const Shape dataShape, Shape constShape) {
-            if ((dataShape.size() - constShape.size()) == 1ul) {
+        auto perChannelQuantization = [](const PartialShape dataPShape, Shape constShape) {
+            if (ngraph::shape_size(constShape) == 1ul) {
+                return true;
+            }
+
+            const auto rank = dataPShape.rank();
+            if (rank.is_dynamic()) {
+                return false;
+            }
+
+            const auto dataShapeSize = static_cast<size_t>(rank.get_length());
+            if ((dataShapeSize - constShape.size()) == 1ul) {
```

Comment:
why only difference of two is supported only?

Reply:
This behavior is reproduced on the master. This method will be reworked as part of another task.